### PR TITLE
fix(Slider): `onUpdate` called on thumb keyboard press, keyboard `onChange` called onKeyUp

### LINKED ISSUE #398: Slider: `onChange` called multiple times during arrow key navigation and `onUpdate` never called
_From https://github.com/iTwin/iTwinUI-react/pull/346#issuecomment-949826671:_

> `onChangeCompleted` is called dozens of times when dragging the dot using arrow keys (instead of only being called on keyup). This happens in both the color square **and the slider**, and it makes the dragging feel laggy especially when combined with a console log.

This behavior should only apply to `onUpdate`, which is instead never called when using keyboard to change values. Contrast to using a mouse, where `onChange` is only called on pointerUp and `onUpdate` is called during dragging.

### Repro

Use arrow left/right keys and observe console in the sandbox. https://codesandbox.io/s/itwinui-react-minimal-example-forked-1q46q?file=/src/App.tsx

### PR DESCRIPTION
**Pull request by r100-stack**
Closes #398 

## Screenshots

* In the below examples, we're pressing keyboard keys: Up/Right, Down/Left, Home, End

#### Old
  * `onUpdate` not called when pressing keyboard keys
  * `onChange` called multiple times when holding keyboard keys

[vokoscreenNG-2022-10-04_15-25-20.webm](https://user-images.githubusercontent.com/45748283/193908360-6900af7b-1f71-48b5-8004-7daaaa5c32a6.webm)

#### New
  * `onUpdate` called even when pressing keyboard keys
  * `onChange` is only called once after the keyboard key  is released

[vokoscreenNG-2022-10-04_16-22-13.webm](https://user-images.githubusercontent.com/45748283/193918989-9c764c87-9bb4-47a9-b671-b81b962ebb3d.webm)

## Checklist

- [x] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- ~~[ ]~~ Add component features demo in Storybook (different stories)
- ~~[ ]~~ Approve test images for new stories
- [x] Add screenshots of the key elements of the component

### CODE REVIEW

File: packages/iTwinUI-react/src/core/Slider/Slider.tsx [hunk 1]
```diff
@@ -334,16 +334,21 @@ export const Slider = React.forwardRef<HTMLDivElement, SliderProps>(
 
     // function called by Thumb keyboard processing
     const onThumbValueChanged = React.useCallback(
-      (index: number, value: number) => {
-        if (currentValues[index] === value) {
+      (index: number, value: number, keyboardReleased: boolean) => {
```
Comment (adamhock) [inline packages/iTwinUI-react/src/core/Slider/Slider.tsx:337]: Minor, but I would change `keyboardReleased` to `keyReleased`

**Review by adamhock — COMMENTED** [review]
It looks good to me at first glance

**Review by mayank99 — APPROVED** [review]
👍 Tested working fine in both Slider and ColorPicker.

Why is this targeting the `v2` branch instead of `main`?

**Comment by rohan-kadkol** [comment]
> +1 Tested working fine in both Slider and ColorPicker.
> 
> Why is this targeting the `v2` branch instead of `main`?

I thought only PR #864 was important and asked by users so that goes to `main` for quick release. But I thought this just a standard priority issue so it could go to `v2` to prevent merge conflicts with `main` in the future.

But if this is a high priority task, I can change it to target `main` too.

**Comment by mayank99** [comment]
> I thought only PR #864 was important and asked by users so that goes to `main` for quick release. But I thought this just a standard priority issue so it could go to `v2` to prevent merge conflicts with `main` in the future.
> 
> But if this is a high priority task, I can change it to target `main` too.

This bug was reported by a user (https://github.com/iTwin/iTwinUI-react/issues/398#issuecomment-1261618819, https://github.com/iTwin/iTwinUI/discussions/769) so it would be nice if we can release a fix without making them wait for v2.

**Comment by rohan-kadkol** [comment]
> > I thought only PR #864 was important and asked by users so that goes to `main` for quick release. But I thought this just a standard priority issue so it could go to `v2` to prevent merge conflicts with `main` in the future.
> > But if this is a high priority task, I can change it to target `main` too.
> 
> This bug was reported by a user ([#398 (comment)](https://github.com/iTwin/iTwinUI-react/issues/398#issuecomment-1261618819), [iTwin/iTwinUI#769](https://github.com/iTwin/iTwinUI/discussions/769)) so it would be nice if we can release a fix without making them wait for v2.

Ohh, right, I forgot about the follow-up comment/request from the user. Sounds good, thanks, changed it to target `main`

**Review by adamhock — APPROVED** [review]
Great work!

**Pull request merged**